services_app/src/main/java/org/opendatakit/services/sync/actions/fragments/SyncFragment.java

```diff
@@ -53,566 +56,606 @@
+            }
+        }
 
-    Intent incomingIntent = getActivity().getIntent();
-    if (savedInstanceState != null && savedInstanceState.containsKey(SYNC_ATTACHMENT_TREATMENT)) {
-      String treatment = savedInstanceState.getString(SYNC_ATTACHMENT_TREATMENT);
-      try {
-        syncAttachmentState = SyncAttachmentState.valueOf(treatment);
-      } catch (IllegalArgumentException e) {
-        syncAttachmentState = SyncAttachmentState.SYNC;
-      }
+        if (savedInstanceState != null && savedInstanceState.containsKey(SYNC_ACTION)) {
+            String action = savedInstanceState.getString(SYNC_ACTION);
+            try {
+                syncAction = SyncActions.valueOf(action);
+            } catch (IllegalArgumentException e) {
+                syncAction = SyncActions.IDLE;
+            }
+        }
+        disableButtons();
     }
 
-    if (savedInstanceState != null && savedInstanceState.containsKey(SYNC_ACTION)) {
-      String action = savedInstanceState.getString(SYNC_ACTION);
-      try {
-        syncAction = SyncActions.valueOf(action);
-      } catch (IllegalArgumentException e) {
-        syncAction = SyncActions.IDLE;
-      }
-    }
-    disableButtons();
-  }
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        super.onCreateView(inflater, container, savedInstanceState);
+        sharedPreferences = this.getActivity().getSharedPreferences(NAME, Context.MODE_PRIVATE);
+        View view = inflater.inflate(ID, container, false);
+
+        infoPane = view.findViewById(R.id.sync_info_pane);
+        populateTextViewMemberVariablesReferences(view);
+        lastSyncField = view.findViewById(R.id.last_sync_field);
+        syncInstanceAttachmentsSpinner = view.findViewById(R.id.sync_instance_attachments);
+
+        if (savedInstanceState != null && savedInstanceState.containsKey(SYNC_ATTACHMENT_TREATMENT)) {
+            String treatment = savedInstanceState.getString(SYNC_ATTACHMENT_TREATMENT);
+            try {
+                syncAttachmentState = SyncAttachmentState.valueOf(treatment);
+            } catch (IllegalArgumentException e) {
+                syncAttachmentState = SyncAttachmentState.SYNC;
+            }
+        }
 
-  @Override public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-    super.onCreateView(inflater, container, savedInstanceState);
+        if (savedInstanceState != null && savedInstanceState.containsKey(SYNC_ACTION)) {
+            String action = savedInstanceState.getString(SYNC_ACTION);
+            try {
+                syncAction = SyncActions.valueOf(action);
+            } catch (IllegalArgumentException e) {
+                syncAction = SyncActions.IDLE;
+            }
+        }
+        lastSyncField.setText(sharedPreferences.getString(LAST_SYNC_INFO_KEY, getResources().getString(R.string.last_sync_not_available)));
```

Comment:
here you can use CommonToolProperties in the following way:

`PropertiesSingleton props  = CommonToolProperties.get(getContext(), getAppName());
 lastSyncField.setText(props.getProperty("LAST_SYNC_INFO_KEY"));
`
you can add a null check to this property when it is not present.